src/framework/ui/view/uitheme.cpp

```diff
@@ -340,14 +397,371 @@ void UiTheme::setupWidgetTheme()
     palette.setColor(QPalette::PlaceholderText, fontPrimaryColor());
     palette.setColor(QPalette::Disabled, QPalette::PlaceholderText, fontPrimaryColorDisabled);
 
+    QApplication::setStyle(this);
     QApplication::setPalette(palette);
 
-    QFont widgetsFont = bodyFont();
-    widgetsFont.setPointSize(configuration()->fontSize(FontSizeType::BODY));
-    QApplication::setFont(widgetsFont);
+    QString styleSheet = QString("*, QSpinBox { font: %1px \"%2\" } ")
+                         .arg(QString::number(bodyFont().pixelSize()), bodyFont().family());
+    qApp->setStyleSheet(styleSheet);
 }
 
 void UiTheme::notifyAboutThemeChanged()
 {
     emit themeChanged();
 }
+
+// ====================================================
+// QStyle
+// ====================================================
+
+void UiTheme::polish(QWidget* widget)
+{
+    QProxyStyle::polish(widget);
+
+    if (qobject_cast<QAbstractItemView*>(widget)
+        || qobject_cast<QGroupBox*>(widget)) {
+        // Make hovering work
+        widget->setMouseTracking(true);
+        widget->setAttribute(Qt::WA_Hover, true);
+    }
+}
+
+void UiTheme::unpolish(QWidget* widget)
+{
+    QProxyStyle::unpolish(widget);
+
+    if (qobject_cast<QAbstractItemView*>(widget)
+        || qobject_cast<QGroupBox*>(widget)) {
+        widget->setMouseTracking(false);
+        widget->setAttribute(Qt::WA_Hover, false);
+    }
+}
+
+void UiTheme::drawPrimitive(QStyle::PrimitiveElement element, const QStyleOption* option, QPainter* painter, const QWidget* widget) const
```

Comment:
This method seems over-complicated for me. I think you can divide it into several small methods

Reply:
Thanks, done!